### PR TITLE
Fix GEOS-7032 for 2.3.2 (still used in 14.01)

### PR DESCRIPTION
**Pull request by pmauduit**
This is an adapted version (and a simpler one, but roughly inspired) of the fix proposed by the GeoServer team. It has been made in a hurry, and some places might have been forgotten, but at least the WFS parts (and the main revealed by the upstream patch IMHO) should have been fixed.

Tested against WFS getfeature operation, before and after this patch, difficult to say if it introduces regressions though.

@jesseeichar I'd love to have at least a quick review on it.

### CODE REVIEW

File: src/community/csw/csw-core/src/main/java/org/geoserver/csw/kvp/GetRecordsKvpRequestReader.java [hunk 1]
```diff
@@ -139,6 +141,7 @@ private QueryType readQuery(Map kvp) throws Exception {
                     Parser parser = new Parser(new OGCConfiguration());
                     parser.setFailOnValidationError(true);
                     parser.setValidating(true);
+                    parser.setEntityResolver(EntityResolverProvider.RESOLVE_DISABLED_PROVIDER);
```
Comment (pmauduit) [inline src/community/csw/csw-core/src/main/java/org/geoserver/csw/kvp/GetRecordsKvpRequestReader.java]: should be getEntityResolver()

File: src/community/csw/csw-core/src/main/java/org/geoserver/csw/xml/v2_0_2/CSWXmlReader.java [hunk 2]
```diff
@@ -40,6 +41,7 @@ public Object read(Object request, Reader reader, Map kvp) throws Exception {
         parser.setValidating(true);
         parser.setFailOnValidationError(true);
         parser.setStrict(true);
+        parser.setEntityResolver(EntityResolverProvider.RESOLVE_DISABLED_PROVIDER);
```
Comment (pmauduit) [inline src/community/csw/csw-core/src/main/java/org/geoserver/csw/xml/v2_0_2/CSWXmlReader.java]: same

File: src/community/wcs2_0/src/main/java/org/geoserver/wcs2_0/xml/WcsXmlReader.java [hunk 3]
```diff
@@ -41,6 +42,7 @@ public Object read(Object request, Reader reader, Map kvp) throws Exception {
         //        parser.setStrict(true);
 
         // parse
+        parser.setEntityResolver(EntityResolverProvider.RESOLVE_DISABLED_PROVIDER);
```
Comment (pmauduit) [inline src/community/wcs2_0/src/main/java/org/geoserver/wcs2_0/xml/WcsXmlReader.java]: same

File: src/extension/wps/wps-core/src/main/java/org/geoserver/wps/xml/v1_0_0/WpsXmlReader.java [hunk 4]
```diff
@@ -38,6 +39,7 @@ public WpsXmlReader(String element, String version, WPSConfiguration configurati
     public Object read(Object request, Reader reader, Map kvp) throws Exception {
         Parser parser = new Parser(configuration);
         parser.setValidating(true);
+        parser.setEntityResolver(EntityResolverProvider.RESOLVE_DISABLED_PROVIDER);
```
Comment (pmauduit) [inline src/extension/wps/wps-core/src/main/java/org/geoserver/wps/xml/v1_0_0/WpsXmlReader.java]: same

**Comment by jesseeichar** [comment]
fine enough

**Pull request merged**